### ISSUE TITLE
Refactor Element DOM structure to improve a11y

### ISSUE THREAD

**Issue by Cheddam** [dnadesign/silverstripe-elemental#697]
Multiple issues [have been found](https://github.com/dnadesign/silverstripe-elemental/issues/396) with the `Element` container in the ElementalEditor:

- The entire `Element` container acts as the open/close toggle for both mouse and keyboard access. Based on this, it has been given a role of `button`, which seems to make it impossible to traverse deeper into the areas within the container.
- The keyboard access also doesn't work at all when using Voiceover's command for pushing the 'button', and instead opens the Actions menu.
- The label applied to the container is read aloud _after_ the entire contents of the container. The characters representing the icons are also read aloud.

Based on this, we should revamp the DOM structure of the `Element` container:

- [ ] Remove the keyboard-based open/close toggle behaviour (retaining the click behaviour)
  - Shift the keyboard control to the expand/collapse icon
- [ ] Use `aria-expanded` / `aria-controls` to designate whether the Element has been opened
  - `aria-controls` should be applied to the expand/collapse icon
- [ ] Use `aria-labelledby` to designate the block title element as what should be announced when the block is selected
- [ ] Add an `sr-only` element to the block title to designate it as a '{type} block'
  - e.g. `<h3 class="element-editor-header__title" id="block-title-2">My Block <span class="sr-only">, Content Block</span></h3>`

**ACs**

- [ ] Ensure that a useful description of an Element is read aloud as soon as it is focussed
- [ ] Ensure that the form elements within an expanded Element container can be navigated to with VoiceOver commands
